packages/web3-providers-ws/test/unit/constructor.test.ts

```diff
@@ -1,10 +1,30 @@
 import Web3ProviderWS from '../../src/index';
-import { WebSocketOptions } from '../../src/types';
 
 describe('Web3ProviderWS Tests', () => {
-    it('should error because invalid WS address', () => {
+    let provider: Web3ProviderWS;
+    it('should throw error for invalid WS address', () => {
         expect(() => {
             new Web3ProviderWS('http://127.0.0.1:8545');
         }).toThrowError('Invalid WebSocket URL provided');
     });
+
+    test('should instantiate with valid client URL', async (done) => {
+        let client = 'ws://127.0.0.1:8545';
+        provider = new Web3ProviderWS(client);
+        expect(provider.web3Client).toBe(client);
+        done();
+    });
+
+    test('should create instance of Web3ProviderWS and call connect', async (done) => {
+        const connectFunc = jest.spyOn(
+            Web3ProviderWS.prototype as any,
+            'connect'
+        );
+        connectFunc.mockImplementation(() => {
+            done();
```

Comment:
Maybe this is the better way to do it, but wouldn't calling `done` here cause the test to finish executing before line 28 executes?

Reply:
thats not the case line 28 is executed, its not in scope of test but in a mock function.

yes its better and mocked here because test is dependent on this  " ...and call connect"